### PR TITLE
Fix #1668 and remove some code

### LINKED ISSUE #1668: New warning in gameworld.cpp
```
[2/4] Building CXX object CMakeFiles/DDNet.dir/src/game/client/prediction/gameworld.cpp.o
src/game/client/prediction/gameworld.cpp: In member function ‘void CGameWorld::CreateExplosion(vec2, int, int, bool, int, int64_t)’:
src/game/client/prediction/gameworld.cpp:305:23: warning: ‘#‘target_mem_ref’ not supported by dump_expr#<expression error>’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   vec2 Diff = apEnts[i]->m_Pos - Pos;
```
@trml

### PR DESCRIPTION
**Pull request by trml**
Also removed the code in #1620 related to the ddnetcharacter freeze flags that will no longer be used.

### CODE REVIEW

File: src/game/client/prediction/gameworld.cpp [hunk 1]
```diff
@@ -296,12 +297,14 @@ CEntity *CGameWorld::GetEntity(int ID, int EntType)
...
 {
 	// deal damage
-	CCharacter *apEnts[MAX_CLIENTS];
+	CCharacter *apEnts[MAX_CLIENTS] = {0};
 	float Radius = 135.0f;
 	float InnerRadius = 48.0f;
 	int Num = FindEntities(Pos, Radius, (CEntity**)apEnts, MAX_CLIENTS, CGameWorld::ENTTYPE_CHARACTER);
 	for(int i = 0; i < Num; i++)
 	{
+		if(!apEnts[i])
```
Comment (def-) [inline src/game/client/prediction/gameworld.cpp]: But could this actually happen? I thought FindEntities only fills non-nullpointers.

Comment (trml) [reply src/game/client/prediction/gameworld.cpp]: Oh, no, the check is actually not needed.

File: src/game/client/prediction/gameworld.cpp [hunk 2]
```diff
@@ -296,7 +297,7 @@ CEntity *CGameWorld::GetEntity(int ID, int EntType)
 void CGameWorld::CreateExplosion(vec2 Pos, int Owner, int Weapon, bool NoDamage, int ActivatedTeam, int64_t Mask)
 {
 	// deal damage
-	CCharacter *apEnts[MAX_CLIENTS];
+	CCharacter *apEnts[MAX_CLIENTS] = {0};
```
Comment (def-) [inline src/game/client/prediction/gameworld.cpp]: Then this is also useless, right?

File: src/game/client/prediction/gameworld.cpp [hunk 3]
```diff
@@ -58,6 +58,8 @@ CEntity *CGameWorld::FindLast(int Type)
 
 int CGameWorld::FindEntities(vec2 Pos, float Radius, CEntity **ppEnts, int Max, int Type)
 {
+	if(!ppEnts)
+		return 0;
```
Comment (def-) [inline src/game/client/prediction/gameworld.cpp]: Why have this?

File: src/game/client/prediction/gameworld.cpp (hunk 2 above)
Comment (trml) [reply src/game/client/prediction/gameworld.cpp]: Yes, but there is a warning without it (which seems to be incorrect). The same code in gamecontext.cpp will for some reason not give a warning.

File: src/game/client/prediction/gameworld.cpp (hunk 3 above)
Comment (trml) [reply src/game/client/prediction/gameworld.cpp]: It isn't really needed either.

**Pull request merged**